### ISSUE TITLE
Proprietary License

### ISSUE THREAD

**Issue by skr4n** [futo-org/grayjay-android#18]
I have already raised the issue with the FUTO Temporary License at [voice-input](https://github.com/futo-org/voice-input/issues/12).
### The custom license

- Grayjay and Voice Input, two of FUTO's great applications are licensed under ["FUTO Temporary License"](https://gitlab.futo.org/alex/voiceinput/-/blob/master/FTL_LICENSE.md?ref_type=heads#section-2-grant-of-rights).
This license does not abide by [the definition of Open Source](https://opensource.org/osd/), by prohibiting modification of source code and redistribution for a fee ([related Youtube comment](https://www.youtube.com/watch?v=5DePDzfyWkw&lc=UgzFgTyvPn1tbdQrX414AaABAg)). Thus this program is NOT Open Source.

### The problem with it
- This clause will prevent a person or an organization from bundling the application as a part of their Custom ROM or a device that they distribute/sell.
- It is misleading to brand the program as Open Source.
- Creating a custom definition of Open Source ([as suggested by Louis Rossmann in a youtube comment](https://www.youtube.com/watch?v=5DePDzfyWkw&lc=UgznX61yDKdnY-NEQB54AaABAg.9vztj1Dt9dE9vzxm-6Y6fI)) doesn't really build rapport with the Open Source community. The term has already been misused many times for Source-available licenses, going further from the true meaning of [Free/Libre software](https://www.gnu.org/philosophy/free-sw.en.html).
- Proprietary licenses don't incentivize contributors.

### FUTO's justification for this license

- FUTO hopes to legally pursue malicious third parties from creating a fork that is ad-infested, trackerware (such as the newpipe situation, where some people forked the project and released a version on the play store filled with ads).

### A solution

- FUTO can utilize their trademarks to pursue any miscreant that release the program under FUTO's brand. 
Libre licenses do not allow trademark infringement.
- This approach is [already used](https://www.sdmmag.com/articles/99966-signal-foundation-sues-adt-over-signal-chat-citing-trademark-infringement) by the [Signal Foundation (AGPLv3)](https://github.com/signalapp/Signal-Android/blob/main/LICENSE) and the [Mozilla Foundation (MPL)](https://github.com/mozilla-mobile/firefox-android/blob/main/LICENSE), both of which are well-established organizations that have produced great software that is totally Libre.

### Conclusion
FUTO is all about giving back control to the user. That is what Libre software grants [with the four freedoms.](https://www.gnu.org/philosophy/free-sw.en.html#four-freedoms)
FUTO has already released great programs such as [Live Captions](https://github.com/abb128/LiveCaptions) under the GPL.
**Thus I, among many others from the Libre / Open Source community suggest FUTO to license the programs as Libre / Open Source software.**

**Comment by syboxez** [comment]
I completely agree. Trademark law already prevents forks from using the original name of the program and would be a good solution to this problem instead of using a more restrictive license.

**Comment by h3ndrik** [comment]
Yes. Trademark law is the correct way to deal with impostors. Not licensing. The statement in the video: "Technically the people who were taking part in that scam did nothing that was illegal" isn't accurate. They maybe violated the trademark of NewPipe and could be held responsible for that. And they probably even violated the GPLv3 if they didn't provide the modified source code with a prominent notice that it was a modified version of NewPipe.

Most of all, **I think the way handling it with a license is probably ineffective. Malicious people probably don't care about the words in your license file. You will mainly restrict the honest users.**

Regarding the statements in [Louis' video](https://www.youtube.com/watch?v=5DePDzfyWkw):

* "We have no problem with that, however if you modify the source code of this application to insert [...] in a deceptive way [...]" Isn't reflected in the written license and thus without substance. Furthermore the wording is probably too vague to make it legally binding in any way.

And there are yet more issues:
* You are severely restricting the users without technical skills. I can probably modify it. But I'm unable to share it then. And the majority of users can't patch software and compile and build it. It is a situation like Unix was in before Linux and specifically the GNU project addressed exactly that.
* non-free software will never get into great projets like the F-Droid store.
* In the domain of YouTube third-party apps, projects like yt-dlp have demonstrated copyleft licensing and the ability to fork them offer additional resilience. It wouldn't be around anymore if it had been source-available only. Grayjay is part of the same dynamics.

That said, it is your project and I didn't code it. So you have the right to choose whatever license you like. But I think you should offer a better issue tracker or engage here and prominently link it in the GitLab instance.

**Comment by HiPhish** [comment]
I wish to point out this article by Richard Stallman: [Did You Say “Intellectual Property”? It's a Seductive Mirage](https://www.gnu.org/philosophy/not-ipr.html)

In the article he argues that the term "intellectual property" conflates three complementary but distinct concepts: trademark, copyright and patent. This part is relevant to the discussion:

> Trademark law, by contrast, was not intended to promote any particular way of acting, but simply to enable buyers to know what they are buying. Legislators under the influence of the term “intellectual property,” however, have turned it into a scheme that provides incentives for advertising. And these are just three out of many laws that the term refers to.
>
>  In practice, nearly all general statements you encounter that are formulated using “intellectual property” will be false. For instance, you'll see claims that “its” purpose is to “promote innovation,” but that only fits patent law and perhaps plant variety monopolies. Copyright law is not concerned with innovation; a pop song or novel is copyrighted even if there is nothing innovative about it. Trademark law is not concerned with innovation; if I start a tea store and call it “rms tea,” that would be a solid trademark even if I sell the same teas in the same way as everyone else. Trade secret law is not concerned with innovation, except tangentially; my list of tea customers would be a trade secret with nothing to do with innovation.

The counterfeit problem is in the domain of trademark and using copyright to fight counterfeiters is the wrong strategy.

**Comment by HiPhish** [comment]
> You are severely restricting the users without technical skills. I can probably modify it. But I'm unable to share it then. And the majority of users can't patch software and compile and build it. It is a situation like Unix was in before Linux and specifically the GNU project addressed exactly that.

I like to call it "maintainer lock-in". Unlike vendor lock-in where no one can make changes, you can *technically* make changes, but only for yourself and as long as you never tell anyone. That's just like saying you can use Grayjay without paying just as long as you never tell anyone.

**Comment by liam-clink** [comment]
You can't call software open source while having no code in the repo... wtf

**Comment by syboxez** [comment]
> You can't call software open source while having no code in the repo... wtf

Source code is here: https://gitlab.futo.org/videostreaming/grayjay

**Comment by junetried** [comment]
The program is open source because the source is open. This is regardless of what GNU or RMS might think about it - who cares. If you're trying to use open source as a buzzword to signal GNU approved licenses then go ahead, but it doesn't belong in a serious GitHub issue IMO.

**Comment by tconnard** [comment]
> The program is open source because the source is open. This is regardless of what GNU or RMS might think about it - who cares. If you're trying to use open source as a buzzword to signal GNU approved licenses then go ahead, but it doesn't belong in a serious GitHub issue IMO.

This isn't about banning trackers or other malware (Discrimination Against Fields of Endeavor), that could be justified even though it would break both OSD and FSD.

Limiting commercial use/distribution is almost by definition proprietary.

**Comment by junetried** [comment]
> > The program is open source because the source is open. This is regardless of what GNU or RMS might think about it - who cares. If you're trying to use open source as a buzzword to signal GNU approved licenses then go ahead, but it doesn't belong in a serious GitHub issue IMO.
> 
> This isn't about banning trackers or other malware (Discrimination Against Fields of Endeavor), that could be justified even though it would break both OSD and FSD.
> 
> Limiting commercial use/distribution is almost by definition proprietary.

You would agree then that the GPL is a proprietary license since it limits distribution?

**Comment by h3ndrik** [comment]
> You would agree then that the GPL is a proprietary license since it limits distribution?

I don't think we need to discuss copyleft vs permissive licenses here. There are Wikipedia articles and several blog posts for that.

Concerning the topic at hand and FUTO, it's about the implications of the license:

[License incompatibility](https://en.wikipedia.org/wiki/License_compatibility) Makes people unable to build upon or combine your stuff. Or incorporate it into the ecosystem of free software projects like [F-Droid](https://f-droid.org) that empower the user.

You shouldn't teach users to sideload APK files from a website. There are severe downsides to that and it can pave the way for some people getting infected by malicious software once they adopt that habit.

You're disproportionately disadvantaging users without technical skills. Most people do not have the technical skills to patch software, compile it and build it for their phones. I can probably modify it. But I'm unable to share it with other people. Granting additional rights to a small percentage of experts, doesn't add anything for your main userbase.
The situation is very similar to the history of early Unix. Back in the days it lead to fragmentation and incompatibility and subsequently the Free Software movment and the GNU project address that.
This sparked the beginnings of the fight for software user freedom. If you advertise your software as 'for the people' you stand in that tradition.

Additional permissions in a video by Lois Rossman are misleading since the written license is binding and that isn't reflected there. It may lead to confusion and the wording is too vague. A license needs to be dependable to be useful to the user.

Coming up with a better license than people worked on for the last 30 years is probably an illusion. In my opinion you should chose one of the well established licenses. It doesn't really matter if it's copyleft or permissive, that is FUTO's choice to make. Just be aware of the consequences.

Other people summed it up better than me:

"A fundamental goal of FOSS is to promote the free exchange of ideas and technology without fear of infringing the rights of others. Ideally, code licensed under like-minded FOSS terms should be freely combinable to create new products. Compatibility is the attribute of software licenses that allows combining of program code. To be compatible, license terms must be free of contradictory or mutually exclusive requirements."

"This unfortunate situation has been exacerbated by the proliferation of incompatible FOSS licenses, many of which differ in only trivial ways. The Open Source Initiative (OSI) was created in part to reduce the fragmentation of the FOSS license space cause by incompatible and redundant licenses. OSI thus strongly encourages using an existing FOSS license instead of creating a new, “bespoke” license, and [offers a categorization of licenses to help developers avoid redundancy](https://opensource.org/licenses-old/category/)."

(from: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3406002/)

**Comment by junetried** [comment]
I agree that it should use a permissive (not copyleft) license, and that it shouldn't blanket-ban derivatives, but to call it not open source is silly. Open source does not imply any license compatibility and there are tons of open source licenses that are incompatible with others. Like the GPL! (not a fan of the GPL)

In fact, even the license itself acknowledges these issues and claims to be a temporary stepping stone to a more permissive license, so these issues are definitely on their radar.

I probably shouldn't be surprised that people here are citing literally RMS to justify this issue, but... And anyway, no one needed to do that, because this can still be an issue without RMS' blessing.

When I heard Louis Rossmann justify the license, my first thought was the zlib/libpng license. It doesn't address all of the concerns he laid out, but I think it's a great permissive license that still allows FUTO to have agency over the Grayjay project branding.

**Comment by h3ndrik** [comment]
> not a fan of the GPL

Mmh. I get that. I'd like people to know more about licensing. We need to be careful here, because sometimes perfect is the enemy of good. But I get what you say. And it's a good idea to read up on this before choosing a license for your work.

> When I heard Louis Rossmann justify the license, my first thought was the zlib/libpng license.

Thanks. I wasn't aware that this imposing the author / needing to rename a fork wasn't in every license. I guess in this case they additionally confused trademark law and licensing.

**Comment by HiPhish** [comment]
I don't know why we are discussing RMS here or discussing permissive VS copyleft. And no, the GPL licenses do not limit distribution. The issue here is false advertising and the consequences of the license FUTO has chosen. Both permissive licenses and copyleft licenses are compatible with the Open Source definition.

Open Source is a term coined by the Open Source Initiative (OSI). Words have definitions, and definitions cannot be self-referential. That means you cannot say "a square is what my buddies and I consider to be a square" and you cannot say "Open Source is what my buddies and I consider to be Open Source". In either cases the term defines itself, which ultimately makes it meaningless. At this point anything is open source. Is minimized Javascript code open source? After all, it is executable Javascript code. Is compiled machine code open source? If you want your own definition, then come up with your own term.

Let's try an analogy. Imagine if Tim Cook announced full Right To Repair support from Apple. You could download the full schematics for any of Apple's devices. However, these schematics would only be for personal use so you can inspect how your MacBooks works. And then Tim Cook went on video yelling into the camera about how if you make changes to anyone's MacBook he will personally come after you. How do you think Louis Rossmann would feel about that? Because that's exactly what he did in his Grayjay video. Sure, Louis could run illegal back-alley MacBook repairs, but I'm pretty sure that's not what he wants, he wants to have a nice honest store in public.

This discussion is pointless. As many others have pointed out, the justification makes no sense because counterfeit applications fall into the domain of trademark law, not copyright law. The points have been made, it is up to FUTO to respond. They can double-down, they can retract, they can discuss the points raised.

**Comment by tconnard** [comment]
There is an advantage to using copyright to enforce misrepresentation: DCMA takedowns.
This is why things such as zlib licence or non-free logos (such as with Firefox) can help.

**Comment by qrpnxz** [comment]
The [GNU General Public License](https://www.gnu.org/licenses/quick-guide-gplv3.html) is the state-of-the-art [copyleft](https://www.gnu.org/licenses/copyleft.html) license. Under the GPL, any work people publish based on Grayjay would have to be released under the same license. It would require they provide all the source that corresponds to that version of the program, even the helper files that let you generate, install, and run that version. (e.g. gradle files)

The Mozilla Public License won't disallow people from releasing alternate versions of the program with malicious extensions that you are not allowed to inspect or otherwise [handle freely](https://www.gnu.org/philosophy/free-sw.html).

I hope Grayjay is allowed to be used freely one day. In that case, I would recommend the GNU GPL because it guarantees those freedoms even when the program is modified.

**Comment by alastairlundy** [comment]
Any update on this?

It's disappointing to see the that Grayjay continues to be advertised by Louis as Open Source despite the fact that it's a lie.

It is source available proprietary software and not what Louis and Futo promised.

As others have mentioned above most developers should not write their own open source license. Everything Louis and Futo are worried about can be fixed by licensing Grayjay under GPLv3 and enforcing the trademark rights of the name and copyright rights of the logo of Grayjay.

**Comment by Danny3** [comment]
Why is the app not available in the main F-droid repository?
Is it fully open source or not?
If I click on the FDroid link from "Also available on  FDroid" I get here:
https://app.futo.org/fdroid/repo/
And if I click on that link, I get here:
https://fdroid.link/#https://app.futo.org/fdroid/repo?fingerprint=39D47869D29CBFCE4691D9F7E6946A7B6D7E6FF4883497E6E675744ECDFA6D6D
Which clearly says as a warning:
"This is a third-party repository. F-Droid does not check apps in this repository for anti-features, trackers, or malware!"
So, what's with the misleading / tricking message of also being available on F-droid, when in fact it's not in the main repository of F-droid?
This is like on Linux, let's say Debian, not being available in its main fully open source repository, but on third party ones, which can include anything as the developers are in complete control instead of the maintainers of that fully open source repository.
There are also no reproducible builds!

Either this comes to F-droid's main repository or that message and link saying that it's also available on F-droid should be removed.
As being available on F-droid, means being in the main repository of F-droid for pretty much everyone, not in third party ones!

**Comment by Danny3** [comment]
As for voice input, this works great as speech to text:
https://f-droid.org/en/packages/org.woheller69.whisperplus/
And it even works with the Heliboard keyboard:
https://f-droid.org/en/packages/helium314.keyboard/
As you can see, both being available in F-droids main repository!

**Comment by Zvonimir-FUTO** [comment]
@Danny3 Grayjay is on our fdroid repository so we don't have to wait for reviews from Fdroid team and we can deploy fixes quickly which is important when platforms break functionality and users don't want to wait additional couple of days for the review process.

All our code is available on our Gitlab instance https://gitlab.futo.org/videostreaming/grayjay/

**Comment by Danny3** [comment]
@Zvonimir-FUTO , your response beats around the bush or what I asked!
Who said that I, as a user, want the absolutely latest fixes and improvements?
As a F-droid (main, default) repository user and as a Debian (main, default) repository user I can wait as much time as needed.
Please don't insult my inteligence as this is the main reason why you refuse to put the app in F-droid's main repository!
If your statement were true, you could easily explain on your website or in the description of the app on F-droid, that you also have a private F-droid repository, fully controlled by you where you can push updates faster, if somebody wants to use that one.
Please stop trying to trick us to think that this is the main reason as long as you can have the app builds in both places and even here on Github.

To make it very clear, I don't trust you and I don't trust any other developer too much, especially with the building process and I want the builds to be created by F-droid maintainers and Debian maintainers, with reproducible builds, like the tens other apps and programs that I use from both repositories.
To me, the third party repositories are not even close in privacy and security to the main default repositories.
I'm glad that I never installed your app on my phone and I will stop installing it on other people's phones too.
I even wanted to create a new issue to ask to start the process of adding it to the Debian's main repository, like I asked other projects, but I know that it will be a straight no, so I don't want to waste my time with that.

Shame that it is this way, I expected more from a person like Louis!

**Comment by HiPhish** [comment]
@Danny3 The FUTO stuff is not being honest here. This is has nothing to do with "getting updates faster". The reason you cannot find this application in the F-Droid repos is because the FUTO license make it neither Free Software (according to the [FSF definition](https://www.gnu.org/philosophy/free-sw.en.html)) nor Open Source Software (according to the [OSI defintion](https://opensource.org/osd)), it is what we commonly call "source available". In particular you cannot distribute modifications to Grayjay, so if the head of FUTO decides on day that there are more fun ways to burn his fortune and the organization shuts down there won't be anyone who is allowed to continue maintenance. The management of FUTO has a very peculiar definition of what qualifies as FOSS.

You can still inspect the source code and build it yourself for your own use, so you don't have to trust the custom F-Droid repository maintainers. Yes, this sucks, it's just like like those "you can repair it for yourself, but you cannot show anyone how you did it" laws.

With that said, not all FUTO software uses the FUTO license. [Immich](https://immich.app/) for example uses the AGPL-3.0 license. Also, Louis Rossmann is no longer with FUTO, and even when he was he was just an employee. I do like what what FUTO is doing, they just need to get their head out of their behind when it comes to licensing.
